### PR TITLE
Fixed Allow Infinite button on Token Approvals

### PR DESCRIPTION
**Pull request by alvarengathomas**
Closes https://github.com/hemilabs/pure.finance/issues/48

This PR includes:
- New command `format:write` on package.json to fix the format of all files;
- Fixed `Allow Infinite` button to be disabled when wallet is not connected;
- Show Infinite Symbol (`∞`) when `Allow Infinite` button is clicked.

Screenshot of the `Allow Infinite` button disabled when wallet is not connected:
Captura de Tela 2024-12-09 às 14 08 20

Screenshot of the Infinite Symbol (`∞`) when `Allow Infinite` button is clicked:
Captura de Tela 2024-12-09 às 14 08 56

### CODE REVIEW

**Review by ArturDolzan — APPROVED** [review]

**Review by gndelia — APPROVED** [review]
utACK

File: site/pages/[locale]/token-approvals.js [hunk 1]
```diff
@@ -14,6 +15,25 @@ import { useTokenInput } from '../../hooks/useTokenInput'
...
+const infiniteSymbol = '∞'
+
+const isInfinite = function (allowance, token) {
+  if (!allowance || !token) {
+    return false
+  }
+
+  if (allowance === infiniteSymbol) {
+    return true
+  }
```
Comment (gabmontes) [inline site/pages/[locale]/token-approvals.js:28]: The logic should not operate on visual/UI state. The infinity symbol shall only be a UI representation of the state of the form. Let's revisit this.

**Review by gabmontes — CHANGES_REQUESTED** [review]

File: site/pages/[locale]/token-approvals.js (hunk 1 above)
Comment (alvarengathomas) [reply site/pages/[locale]/token-approvals.js]: fixed, using component state `isInfinity` now

**Review by ArturDolzan — DISMISSED** [review]

File: site/pages/[locale]/token-approvals.js [hunk 2]
```diff
@@ -61,11 +57,8 @@ const useAllowanceInput = function (
       tokenApprovals
         .allowance(token.address, account, spender.address)
         .then(function (currentAllowance) {
-          setAllowance(
-            isInfinite(currentAllowance, token)
-              ? infiniteSymbol
-              : fromUnit(currentAllowance, token.decimals)
-          )
+          ;``
```
Comment (gndelia) [inline site/pages/[locale]/token-approvals.js:60]: This line should be removed

Comment (alvarengathomas) [reply site/pages/[locale]/token-approvals.js]: removed

**Review by gndelia — APPROVED** [review]
utACK

**Review by ArturDolzan — APPROVED** [review]

**Review by gabmontes — APPROVED** [review]
uACK

**Pull request merged**